src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDriver.java

```diff
@@ -731,7 +731,33 @@ public final class SQLServerDriver implements java.sql.Driver {
     static final String AUTH_DLL_NAME = "mssql-jdbc_auth-" + SQLJdbcVersion.MAJOR + "." + SQLJdbcVersion.MINOR + "."
             + SQLJdbcVersion.PATCH + "." + Util.getJVMArchOnWindows() + SQLJdbcVersion.RELEASE_EXT;
     static final String DEFAULT_APP_NAME = "Microsoft JDBC Driver for SQL Server";
+    static final String APP_NAME_TEMPLATE = "Microsoft JDBC - %s, %s - %s";
 
+    /**
+     * Constructs the application name using system properties for OS, platform, and architecture.
+     * If any of the properties cannot be fetched, it falls back to the default application name.
+     * Format -> Microsoft JDBC - {OS}, {Platform} - {architecture}
+     *
+     * @return the constructed application name or the default application name if properties are not available
+     */
+    static String getAppName() {
+        String osName = System.getProperty("os.name", "");
+        String osArch = System.getProperty("os.arch", "");
+        String javaVmName = System.getProperty("java.vm.name", "");
+        String javaVmVersion = System.getProperty("java.vm.version", "");
+
+        String platform = javaVmName.isEmpty() || javaVmVersion.isEmpty() ? "" : javaVmName + " " + javaVmVersion;
+
+        String appName = String.format(APP_NAME_TEMPLATE, osName, platform, osArch);
+        
+        // If all values are empty, return the default app name
+        if (osName.isEmpty() && platform.isEmpty() && osArch.isEmpty()) {
+            return DEFAULT_APP_NAME;
```

Comment:
If osName, platform and osArch are empty, then we can avoid String formatting: 
   
  if (!osName.isEmpty() && !platform.isEmpty() && !osArch.isEmpty()) {
      return  String.format(APP_NAME_TEMPLATE, osName, platform, osArch); 
  }
  return DEFAULT_APP_NAME;

Reply:
this should be a static and only done once